src/components/Swap/Toolbar/Toolbar.test.tsx

```diff
@@ -0,0 +1,72 @@
+import { TradeType } from '@uniswap/sdk-core'
+import { DAI, USDC_MAINNET } from 'constants/tokens'
+import { SwapInfoProvider } from 'hooks/swap/useSwapInfo'
+import { flagsAtom } from 'hooks/useSyncFlags'
+import Module from 'module'
+import { Field, stateAtom, Swap } from 'state/swap'
+import { renderComponent } from 'test'
+
+import Toolbar from './index'
+
+jest.mock('@web3-react/core', () => ({
+  ...(jest.requireActual('@web3-react/core') as Module),
+  useWeb3React: () => ({
+    isActive: true,
+    chainId: 1,
+    account: '0x0000000000000000000000000000000000000000',
+  }),
+}))
+
+jest.mock('../../../hooks/usePermit2Allowance', () => ({
+  __esModule: true,
+  ...(jest.requireActual('../../../hooks/usePermit2Allowance') as Module),
+  default: () => ({
+    state: 1 /* AllowanceState.REQUIRED */,
+    isApprovalLoading: false,
+    approveAndPermit: jest.fn(),
+  }),
+}))
+
+function getInitialTradeState(trade: Partial<Swap> = {}) {
+  return {
+    type: TradeType.EXACT_INPUT,
+    amount: '',
+    [Field.INPUT]: DAI,
+    [Field.OUTPUT]: USDC_MAINNET,
+    ...trade,
+  }
+}
+
+describe('Toolbar', () => {
+  it('should not render without currencies specified', () => {
+    const component = renderComponent(<Toolbar />)
+    expect(component.queryByTestId('toolbar')).toBeNull()
+  })
+
+  it('should render with both currencies specified, no amount input', () => {
+    const component = renderComponent(
+      <SwapInfoProvider>
```

Comment:
Should this provider be auto-included and mocked (as an additional option) as part of `renderComponent`/`renderHook`?

Reply:
This could be done as a follow-up. I'm more interested in whether you think this would make future tests easier to write. If so, I'd be happy to own this piece.

Reply:
i think it would make sense to move this wrapping of the UI under test into the "test infra" section of the code. we could have an API that the test writer uses to enable different providers for wrapping lower level components, if needed. although, i'm not sure if there is any harm to always wrapping ...

Reply:
i don't think it's urgent until we get to the point where we're doing the same thing in multiple test files and copy-pasting between them just to write tests